runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaTimerInternalsFactory.java

```diff
@@ -191,11 +193,26 @@ public Collection<KeyedTimerData<K>> removeReadyTimers() {
         state.reloadEventTimeTimers();
       }
     }
+
+    // Flush all timers for process time
+    final Iterator<KeyedTimerData<K>> processBufferIterator = processTimeBuffer.iterator();
+    while (processBufferIterator.hasNext() && readyTimers.size() < maxReadyTimersToProcessOnce) {
+      KeyedTimerData<K> processTimerData = processBufferIterator.next();
+      readyTimers.add(processTimerData);
+      processBufferIterator.remove();
+      state.deletePersisted(processTimerData);
```

Comment:
Given we persist the deletes for the process timers, are the ready timers persisted themselves in case of failures?

Reply:
due to samza checkpointing with changelog/blobstore, and state on restart from failure is read from the checkpoint any persisted deletes will be rolled back